query.go

```diff
@@ -260,17 +376,36 @@ func (q *query) queryPeer(p peer.ID) *queryResult {
 
 		// add their addresses to the dialer's peerstore
 		q.dht.peerstore.AddAddrs(next.ID, next.Addrs, pstore.TempAddrTTL)
-		closer := q.localPeers.Add(next.ID)
-		foundCloserPeer = foundCloserPeer || closer
+		saw = append(saw, next.ID)
 	}
 
-	// did the successful query RPC fulfill the query stop condition ?
-	if q.stopFn(q.localPeers) {
-		q.cancel()
-	}
+	ch <- &queryUpdate{seen: saw, queried: []peer.ID{p}}
+}
 
-	return &queryResult{
-		foundCloserPeer: foundCloserPeer,
+func (q *query) updateState(up *queryUpdate) {
```

Comment:
Yes, I left them in pending comments from @petar since he was the one who advocated for them being left in defensively. I'm inclined to agree that we should be more closely controlling state transitions and logging errors though.